tests/common.go

```diff
@@ -7581,3 +7581,182 @@ func GetClusterProviders() []string {
+	output = strings.TrimSpace(output)
+
+	num, err := strconv.ParseUint(output, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return num, nil
+}
+
+// IsIORunningOnVolume return true if IO is running on the volume
+func IsIORunningOnVolume(vol volume.Volume) (bool, error) {
+	log.Infof("Checking if io is running on the volume [%v]", vol.Name)
+	inspectVol, err := GetVolumeConsumedSize(vol)
+	if err != nil {
+		return false, err
+	}
+	usageBefore := inspectVol
+
+	// Sleep for 10 sec before reading back the output of the volume
+	time.Sleep(10 * time.Second)
+
+	inspectVol, err = GetVolumeConsumedSize(vol)
+	if err != nil {
+		return false, err
+	}
+
+	usageAfter := inspectVol
+	if usageBefore != usageAfter {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// GetAllVolumesWithIO Returns list of volumes with IO
+func GetAllVolumesWithIO(contexts []*scheduler.Context) ([]*volume.Volume, error) {
+
+	allVolsWithIO := []*volume.Volume{}
+	for _, eachContext := range contexts {
+		vols, err := Inst().S.GetVolumes(eachContext)
+		if err != nil {
+			log.Errorf("Failed to get app %s's volumes", eachContext.App.Key)
+		}
+		log.Infof("list of all volumes present in the cluster [%v]", vols)
+		for _, eachVol := range vols {
+			VolStatus, err := IsIORunningOnVolume(*eachVol)
+			if err != nil {
+				return nil, err
+			}
+
+			if VolStatus {
+				allVolsWithIO = append(allVolsWithIO, eachVol)
+			}
+		}
+	}
+	return allVolsWithIO, nil
+}
+
+func IsVolumeFull(vol volume.Volume) (bool, error) {
+	volPercentage, err := GetVolumeFullPercentage(vol)
+	if err != nil {
+		return false, err
+	}
+	if volPercentage > 80.0 {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// GetVolumeFullPercentage returns percentage of volume size consumed
+func GetVolumeFullPercentage(vol volume.Volume) (float64, error) {
+	calculatePercentage := func(usedSize float64, totalSize float64) float64 {
+		percentage := (usedSize / totalSize) * 100
+		return float64(percentage)
+	}
+
+	appVol, err := Inst().V.InspectVolume(vol.ID)
+	log.FailOnError(err, fmt.Sprintf("err inspecting vol : %s", vol.ID))
+
+	totalSize := appVol.Spec.Size
+	usedSize, err := GetVolumeConsumedSize(vol)
+	if err != nil {
+		return 0, err
+	}
+
+	log.Infof("Total size of Volume is [%v] and used size is [%v]", totalSize, usedSize)
+	percentageFull := calculatePercentage(float64(usedSize), float64(totalSize))
+	log.Infof("Volume [%v] : Percentage of size consumed [%v]", vol.Name, percentageFull)
+
+	return percentageFull, nil
+}
+
+// IsStorageNodeDown returns true if storage node is down
+func IsStorageNodeDown(n node.Node) (bool, error) {
+	nodeStatus, err := Inst().V.GetNodeStatus(n)
+	if err != nil {
+		return false, err
+	}
+	if *nodeStatus == opsapi.Status_STATUS_STORAGE_DOWN {
+		log.InfoD("Status of the node is [%v]", nodeStatus)
+		return true, nil
+	}
+	return false, nil
+}
+
+// GetPoolCapacityUsed Get Pool capacity percentage used
+func GetPoolCapacityUsed(poolUUID string) (float64, error) {
```

Comment:
delete if not required

Reply:
this is part of library , but not currently in use. this is to get the fill percentage of the pool .